file.go

```diff
@@ -10,14 +14,104 @@ import (
 var (
 	_ fs.File     = (*file)(nil)
 	_ fs.FileInfo = (*fileInfo)(nil)
+	_ io.Seeker   = (*file)(nil)
 )
 
 type file struct {
-	io.ReadCloser
-	stat func() (fs.FileInfo, error)
+	cl     s3iface.S3API
+	bucket string
+	name   string
+
+	realReader      io.ReadCloser
+	stat            func() (fs.FileInfo, error)
+	currentPosition int64
```

Comment:
please rename to `offset` or `i` for index

Reply:
renamed